src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/DamageCommas.java

```diff
@@ -50,10 +52,13 @@ public class DamageCommas {
 	private static final Pattern PATTERN_NO_CRIT = Pattern.compile("(\u00a7.)([\\d+,]*)(.*)");
 	private static final Pattern OVERLOAD_PATTERN = Pattern.compile("(\u00a7.)" + OVERLOAD_STAR + "((?:\u00a7.[\\d,])+)(\u00a7.)" + OVERLOAD_STAR + "\u00a7r");
 
+	private static Map<String, IChatComponent> nameCache = new HashMap<>();
+
 	public static IChatComponent replaceName(EntityLivingBase entity) {
-		if (!entity.hasCustomName()) return entity.getDisplayName();
+		String rawName = entity.getName();
+		IChatComponent name = Utils.getOrPut(nameCache, rawName, entity::getDisplayName);
+		if (!entity.hasCustomName()) return name;
```

Comment:
Fairly sure this will just create more performance issues? It will increase RAM usage *and* it will decrease raw CPU performance.

Reply:
fair point.
Should we rather avoid touching IChatComponent here then? as this still creates new objects on every call, filling up the gc quickly.